website/blog/2018-06-25-on-consuming-and-publishing-es6+-packages.md

```diff
@@ -0,0 +1,218 @@
+
+I feel like this is similar to how we had a Babel preset for each year and ended up creating `preset-latest` which included all the yearly plugins (we removed that soon after in favor of `preset-env`).
+
+Having a field like `es-latest` or `esnext` may not be helpful either because we want the library versions to be fixed yet changing. And providing only the source may not always be helpful if you use non-standard syntax, TS, Flow, JSX, etc which might have different behavior based on compiler version. Maybe we need some changes on the `npm` side?
+
+One suggestion is to provide an ES5 version but also publish a version that includes the latest standard syntax so it can be compiled with `preset-env`.
+
+### Dependencies May Not Publish ES6+
+
+TODO:
+
+- Due to complexity and tooling, it may be difficult for projects to publish ES6/ESM without more setup. This is probably the biggest issue to get right, even docs aren't enough. We should add some feature requests to `@babel/cli` to make this easier, and maybe make the `babel` package do this by default? Tools like @developit's [microbundle](https://github.com/developit/microbundle) may help a lot with this.
+- How do we deal with polyfills? This could be it's own post (I never finished it). Mention `preset-env` + polyfills. What would it look like for a library author not to have to think about polyfills (or the user).
+
+### Compiler Complexity
+
+Every compiler has bugs, although the risk is lower if there are so many users and sufficient tests. We should just be aware that compiling dependencies does increase the surface area of potential bugs, but I don't believe that should deter us from making this a common practice.
+
+- Babel v6 assumed everything that it compiled was a module and thus in strict mode (one could argue this is actually a good thing). Thus if you tried to run Babel on all your `node_modules` and it encountered something that was a `script` like a jQuery plugin it might cause a lot of issues. This is changed in v7 so that it won't always auto inject the `"use strict"` directive unless it actually is a module.
+- React Native has always compiled `node_modules` by default and has probably learned a lot from issues like ^.
+- `preset-env` could have its own bugs because we use `compat-table` vs. test262.
+- Browsers themselves can have issues with running native ES6+ code vs. ES5.
+- Question of how do we determine what is "supported": https://github.com/babel/babel-preset-env/issues/54 example of edge case.
+
+### Help with Babel v7 Config
+
+It can be pretty painful to compile `node_modules` in Babel v6. The reason is both that we never thought of it as something we wanted as well as getting issue reports that people would actually accidently do it making the build slower.
```

Comment:
Move the second sentence to where you made this point already in the "Compiler complexity" section.

As an intro to this section, you want to establish what Babel v7 changes. I'd suggest something like:

> As I've discussed, compiling dependencies in Babel v6 can be pretty painful. Babel v7 will address some of these pain points.